usecases/monitoring/prometheus.go

```diff
@@ -141,50 +145,102 @@ type PrometheusMetrics struct {
 	T2VRequestsPerBatch   *prometheus.HistogramVec
 }
 
-func NewServerMetrics(namespace string, reg prometheus.Registerer) *ServerMetrics {
+// NewHTPServerMetrics return the ServerMetrics that can be used in any of the grpc or http servers.
+//
+// Example for HTTP servers.
+//
+//	weaviate_request_duration_seconds_sum{instance="node-1", method="DELETE", route="/v1/schema/{className}", status_code="200"}
+//
+// Example For GRPC servers
+//
+//	weaviate_request_duration_seconds_sum{instance="node-1", method="gRPC", route="/grpc.health.v1.Health/Check", status_code="OK"}
+func NewHTTPServerMetrics(namespace string, reg prometheus.Registerer) *HTTPServerMetrics {
 	r := promauto.With(reg)
+	subsystem := "http"
 
-	return &ServerMetrics{
-		TCPActiveConnections: r.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "tcp_active_connections",
-			Help:      "Current number of accepted TCP connections.",
-		}, []string{"protocol"}),
+	return &HTTPServerMetrics{
 		RequestDuration: r.NewHistogramVec(prometheus.HistogramOpts{
 			Namespace: namespace,
+			Subsystem: subsystem,
 			Name:      "request_duration_seconds",
-			Help:      "Time (in seconds) spent serving HTTP requests.",
+			Help:      "Time (in seconds) spent serving requests.",
 			Buckets:   LatencyBuckets,
 		}, []string{"method", "route", "status_code"}),
 		RequestBodySize: r.NewHistogramVec(prometheus.HistogramOpts{
 			Namespace: namespace,
+			Subsystem: subsystem,
 			Name:      "request_message_bytes",
```

Comment:
If we don't want to use the library, I would suggest to at least follow the names that it is using since they are quite standard: https://github.com/slok/go-http-metrics/blob/master/metrics/prometheus/prometheus.go#L73

the term `message` is not used In the context of HTTP